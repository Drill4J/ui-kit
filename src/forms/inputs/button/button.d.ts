import { ButtonType } from './button-type';
export declare const Button: DOMComponent<ButtonType>;
