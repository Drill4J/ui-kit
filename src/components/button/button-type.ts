export interface ButtonType {
  type?: 'primary' | 'secondary';
  size?: 'large' | 'small';
  disabled?: boolean;
  'data-test'?: string;
}
