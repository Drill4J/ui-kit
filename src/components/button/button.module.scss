@import '../../style-guide/index';

.button {
  border-radius: 20px;
  outline: 0;
  cursor: pointer;

  & > :not(:last-child) {
    margin-right: 8px;
  }

  &--type {
    &--primary {
      border: 1px solid map-get($color-palette, blue100);
      background-color: map-get($color-palette, blue100);
      color: map-get($color-palette, white);
      font-family: $semibold-font;

      &:hover {
        border: 1px solid map-get($color-palette, blue75);
        background-color: map-get($color-palette, blue75);
      }

      &:active {
        border: 1px solid map-get($color-palette, blue125);
        background-color: map-get($color-palette, blue125);
      }
    }
    &--secondary {
      border: 1px solid map-get($color-palette, blue100);
      background-color: map-get($color-palette, white);
      color: map-get($color-palette, blue100);
      font-family: $semibold-font;

      &:hover {
        border: 1px solid map-get($color-palette, blue75);
      }

      &:active {
        border: 1px solid map-get($color-palette, blue125);
      }
    }
  }

  &--size {
    &--large {
      height: 32px;
      padding: 6px 16px;
      font-size: 14px;
    }

    &--small {
      height: 24px;
      padding: 1px 12px;
      font-size: 12px;
    }
  }

  &--disabled {
    opacity: 0.25;
    pointer-events: none;
  }
}
