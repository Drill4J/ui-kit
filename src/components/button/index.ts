export { Button } from './button';
export { CancelButton } from './cancel-button';
export { LinkButton } from './link-button';
