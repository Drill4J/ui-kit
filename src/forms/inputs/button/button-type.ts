export interface ButtonType {
  kind?: 'primary' | 'secondary';
  size?: 'large' | 'small';
  disabled?: boolean;
  'data-test'?: string;
}
