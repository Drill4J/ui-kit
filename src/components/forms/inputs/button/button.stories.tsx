import { storiesOf } from '@storybook/react';
import 'twin.macro';

import { Button } from './button';
import { CancelButton } from './cancel-button';
import { LinkButton } from './link-button';
import { NegativeActionButton } from './negative-action-button';

storiesOf('Button', module).add('types of buttons', () => (
  <div tw="space-y-5">
    <Button primary size="large">
      Primary large
    </Button>
    <Button primary size="small">
      Primary small
    </Button>
    <Button secondary size="large">
      Secondary large
    </Button>
    <Button secondary size="small">
      Secondary small
    </Button>
    <Button primary size="large">
      <span>Primary large</span>
    </Button>
    <Button secondary size="small">
      <span>Secondary small</span>
    </Button>
    <CancelButton size="large">Cancel button large</CancelButton>
    <CancelButton size="small">Cancel button small</CancelButton>
    <div tw="flex flex-col">
      <LinkButton size="large">Link button large</LinkButton>
      <LinkButton size="small">Link button small</LinkButton>
    </div>
    <NegativeActionButton size="large">Negative action button large</NegativeActionButton>
    <NegativeActionButton size="small">Negative action button small</NegativeActionButton>
  </div>
));
