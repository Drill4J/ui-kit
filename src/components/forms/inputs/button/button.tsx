import tw, { styled } from 'twin.macro';

import { ButtonType } from './button-type';

export const Button = styled.button<ButtonType>`
  ${tw`flex items-center rounded-2xl gap-x-2 outline-none cursor-pointer`}
  
  ${({
    primary, secondary, size, disabled,
  }) => [
    primary &&
    tw`border border-blue-default bg-blue-default text-monochrome-white font-bold
      hover:border-blue-medium-tint hover:bg-blue-medium-tint
      active:border-blue-shade active:bg-blue-shade 
    `,
    secondary &&
    tw`border border-blue-default bg-transparent text-blue-default font-bold
      hover:border-blue-medium-tint
      active:border-blue-shade
    `,
    size === 'small' && tw`h-6 leading-16 py-0 px-3 text-12`,
    size === 'large' && tw`h-8 leading-20 py-0 px-4 text-14`,
    disabled && tw`opacity[0.4] pointer-events-none`,
  ]}
`;
