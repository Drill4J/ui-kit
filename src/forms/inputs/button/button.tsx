import styled, { css } from 'styled-components';

import { ButtonType } from './button-type';
import { COLORS, FONTS } from '../../../theme';

export const Button = styled.button<ButtonType>`
  display: flex;
  align-items: center;
  border-radius: 16px;
  outline: 0;
  cursor: pointer;
  
  ${({ kind, size, disabled }) => [
    kind === 'primary' && css`
      border: 1px solid ${COLORS.PRIMARY_BLUE.DEFAULT};
      background-color: ${COLORS.PRIMARY_BLUE.DEFAULT};
      color: ${COLORS.MONOCHROME.WHITE};
      font-family: ${FONTS.SEMI_BOLD};

      &:hover {
        border: 1px solid ${COLORS.PRIMARY_BLUE.MEDIUM_TINT};
        background-color: ${COLORS.PRIMARY_BLUE.MEDIUM_TINT};
      }

      &:active {
        border: 1px solid ${COLORS.PRIMARY_BLUE.SHADE};
        background-color: ${COLORS.PRIMARY_BLUE.SHADE};
      }
    `,
    kind === 'secondary' && css`
      border: 1px solid ${COLORS.PRIMARY_BLUE.DEFAULT};
      background-color: ${COLORS.MONOCHROME.WHITE};
      color: ${COLORS.PRIMARY_BLUE.DEFAULT};
      font-family: ${FONTS.SEMI_BOLD};

      &:hover {
        border: 1px solid ${COLORS.PRIMARY_BLUE.MEDIUM_TINT};
      }

      &:active {
        border: 1px solid ${COLORS.PRIMARY_BLUE.SHADE};
      } 
    `,
    size === 'small' && css`
      height: 24px;
      line-height: 16px;
      padding: 0 12px;
      font-size: 12px; 
    `,
    size === 'large' && css`
      height: 32px;
      line-height: 20px;
      padding: 0 16px;
      font-size: 14px;
    `,
    disabled && css`
      opacity: 0.4;
      pointer-events: none;  
    `,
  ]}
`;
