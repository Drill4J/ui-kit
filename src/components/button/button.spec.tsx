import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from './button';

it('renders correctly', () => {
  const tree = renderer.create(<Button type="primary" size="large">Primary large</Button>).toJSON();
  
  expect(tree).toMatchSnapshot();
});
