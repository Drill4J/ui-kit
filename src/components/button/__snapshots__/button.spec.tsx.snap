// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button button--type--primary button--size--large"
  data-test=""
  disabled={false}
  onClick={[Function]}
  size="large"
  type="primary"
>
  Primary large
</button>
`;
